containerd-shim-spin/src/engine.rs

```diff
@@ -198,7 +198,22 @@ impl SpinEngine {
             }
         };
         info!(" >>> notifying main thread we are about to start");
-        f.await
+        let (abortable, abort_handle) = futures::future::abortable(f);
+        ctrlc::set_handler(move || abort_handle.abort())?;
+        match abortable.await {
+            Ok(Ok(())) => {
+                info!("Trigger executor shut down: exiting");
+                Ok(())
+            }
+            Ok(Err(err)) => {
+                log::error!("ERROR >>> Trigger executor failed: {:?}", err);
+                Err(err)
+            }
+            Err(aborted) => {
```

Comment:
Hi @Mossaka, 

My understanding (based on running tests with a k3d based setup) is that when a pod is terminating, it will send the TERM signal to the shim. and because we were not handling the term signal, this will continue to run until a KILL signal is sent after TerminateGracefulSeconds limit is crossed.

Thank you